### PR TITLE
Add new BSD Variant with tests #859

### LINKED ISSUE #859: Add new BSD variant
See https://lists.spdx.org/pipermail/spdx-legal/2017-November/002397.html
Reported to SPDX by @pgiffuni
Fir now it is only detected as a partial BSD simplified with a low 78% coverage of the 2 clause BSD:
```
{
  "scancode_notice": "Generated with ScanCode and provided on an \"AS IS\" BASIS, WITHOUT WARRANTIES\nOR CONDITIONS OF ANY KIND, either express or implied. No content created from\nScanCode should be considered or used as legal advice. Consult an Attorney\nfor any legal advice.\nScanCode is a free software code scanning tool from nexB Inc. and others.\nVisit https://github.com/nexB/scancode-toolkit/ for support and download.",
  "scancode_version": "2.2.1.post98.5682f7a3.dirty.20171117195748",
  "scancode_options": {
    "--license": true,
    "--license-score": 0,
    "--license-text": true,
    "--format": "json-pp",
    "--diag": true
  },
  "files_count": 1,
  "files": [
    {
      "path": "scancode-toolkit-master/t",
      "scan_errors": [],
      "licenses": [
        {
          "key": "bsd-simplified",
          "score": 78.02,
          "short_name": "BSD-Simplified",
          "category": "Permissive",
          "owner": "Regents of the University of California",
          "homepage_url": "http://www.opensource.org/licenses/BSD-2-Clause",
          "text_url": "http://opensource.org/licenses/bsd-license.php",
          "reference_url": "https://enterprise.dejacode.com/urn/urn:dje:license:bsd-simplified",
          "spdx_license_key": "BSD-2-Clause",
          "spdx_url": "https://spdx.org/licenses/BSD-2-Clause",
          "start_line": 1,
          "end_line": 17,
          "matched_rule": {
            "identifier": "bsd-simplified_35.RULE",
            "license_choice": false,
            "licenses": [
              "bsd-simplified"
            ],
            "matcher": "3-seq",
            "rule_length": 182,
            "matched_length": 142,
            "match_coverage": 78.02,
            "rule_relevance": 100
          },
          "matched_text": "Redistribution and use in source and binary forms, with or without\nmodification, are permitted provided that the following conditions\nare met:\n1. Redistributions of source code must retain the above copyright\n   notice, this list of conditions and the following disclaimer.\n\n[THIS] [SOFTWARE] [IS] [PROVIDED] [BY] [[Name] [of] [Copyright] [Owner]], Inc. ``AS IS'' AND\nANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE\nIMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE\nARE DISCLAIMED.  IN NO EVENT SHALL [[Name] [of] [Copyright] [Owner]] BE LIABLE\nFOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL\nDAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS\nOR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)\nHOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT\nLIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY\nOUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF\nSUCH DAMAGE"
        }
      ]
    }
  ]
}
```

### PR DESCRIPTION
**Pull request by aviral1701**
Fixes #859

### CODE REVIEW

**Comment by pombredanne** [comment]
Thanks! let me review this.

File: src/licensedcode/data/licenses/bsd-1-clause.LICENSE [hunk 1]
```diff
@@ -0,0 +1,22 @@
+BSD 1 Clause License
+
+Copyright (c) [Copyright year]
+[Name of Copyright Owner]  All rights reserved.
```
Comment (pombredanne) [inline src/licensedcode/data/licenses/bsd-1-clause.LICENSE]: We never put copyrights in license texts, especially not templates

File: tests/licensedcode/data/licenses/bsd-1-clause.txt [hunk 2]
```diff
@@ -0,0 +1,22 @@
+BSD 1 Clause License
+
+Copyright (c) [Copyright year]
+[Name of Copyright Owner]  All rights reserved.
+
```
Comment (pombredanne) [inline tests/licensedcode/data/licenses/bsd-1-clause.txt]: ...but this is fine for the tests of course!

File: tests/licensedcode/data/licenses/bsd-1-clause.yml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+licenses:
+  - bsd-1-clause
```
Comment (pombredanne) [inline tests/licensedcode/data/licenses/bsd-1-clause.yml]: can you add a line feed?

**Comment by pombredanne** [comment]
Also I am not sure why tests are failing: do they pass on your machine?

File: src/licensedcode/data/licenses/bsd-1-clause.LICENSE (hunk 1 above)
Comment (aviral1701) [reply src/licensedcode/data/licenses/bsd-1-clause.LICENSE]: Will take care.

**Comment by aviral1701** [comment]
Nope,I couldn't get tests to run on my machine.I'll make the edits and try again.Update you soon.

**Comment by aviral1701** [comment]
@pombredanne  I have made the changes.Please review.AIso I still couldn't get the tests to run on my machine.I get the following error:

```
* Building license index...
Traceback (most recent call last):
  File "/home/d3xt3r/scancode-toolkit/etc/conf/base.py", line 59, in <module>
    build_license_cache()
  File "/home/d3xt3r/scancode-toolkit/etc/conf/base.py", line 56, in build_license_cache
    cache.reindex()
  File "/home/d3xt3r/scancode-toolkit/src/licensedcode/cache.py", line 145, in get_or_build_index_through_cache
    rules_data_dir=_rules_data_dir)
  File "/home/d3xt3r/scancode-toolkit/src/licensedcode/models.py", line 452, in get_rules
    licenses = get_licenses_db(licenses_data_dir=licenses_data_dir)
  File "/home/d3xt3r/scancode-toolkit/src/licensedcode/cache.py", line 206, in get_licenses_db
    _LICENSES = load_licenses(licenses_data_dir)
  File "/home/d3xt3r/scancode-toolkit/src/licensedcode/models.py", line 438, in load_licenses
    lic = License(key, licenses_data_dir)
  File "/home/d3xt3r/scancode-toolkit/src/licensedcode/models.py", line 150, in __init__
    self.load(src_dir)
  File "/home/d3xt3r/scancode-toolkit/src/licensedcode/models.py", line 301, in load
    setattr(self, k, v)
AttributeError: 'License' object has no attribute 'Key'

Failed to execute command:
/home/xxx/scancode-toolkit/bin/python "/home/xxx/scancode-toolkit/etc/conf/base.py". Aborting...
```
Any leads?
The tests are failing as well.

**Comment by pombredanne** [comment]
Do you mind to squash these two commit in one proper commit with a good commit message? that would be great!

File: src/licensedcode/data/licenses/bsd-1-clause.yml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+Key: bsd-1-clause
```
Comment (pombredanne) [inline src/licensedcode/data/licenses/bsd-1-clause.yml]: `key` is always lowercase

**Comment by aviral1701** [comment]
sure! will update in a couple hours.

File: src/licensedcode/data/licenses/bsd-1-clause.yml (hunk 4 above)
Comment (aviral1701) [reply src/licensedcode/data/licenses/bsd-1-clause.yml]: ok.

**Comment by aviral1701** [comment]
@pombredanne Please review this.Thanks

**Comment by pombredanne** [comment]
Done and merged

**Comment by aviral1701** [comment]
Thanks for all the help. @pombredanne

**Pull request merged**